### PR TITLE
[ActionSheet] Add `contentEdgeInsets` API.

### PR DESCRIPTION
**Pull request by romoore**
This is the first of three APIs required to support customized action cell
layouts necessary for internal design guidelines.  The `contentEdgeInsets` API
allows configuring the inset/outset values of the content rectangle of each
action cell.

Part of #8709

### CODE REVIEW

File: components/ActionSheet/src/private/MDCActionSheetItemTableViewCell.m [hunk 1]
```diff
@@ -67,19 +88,21 @@ - (void)commonMDCActionSheetItemViewInit {
...
-  [_actionLabel.bottomAnchor constraintEqualToAnchor:self.contentView.bottomAnchor
-                                            constant:-kActionItemTitleVerticalPadding]
-      .active = YES;
-  _titleLeadingConstraint = [_actionLabel.leadingAnchor
-      constraintEqualToAnchor:self.contentView.layoutMarginsGuide.leadingAnchor
-                     constant:leadingConstant];
+  NSLayoutConstraint *labelBottomConstraint =
+      [_actionLabel.bottomAnchor constraintEqualToAnchor:_contentContainerView.bottomAnchor
+                                                constant:-kActionItemTitleVerticalPadding];
+  labelBottomConstraint.priority = UILayoutPriorityDefaultHigh;
```
Comment (codeman7) [inline components/ActionSheet/src/private/MDCActionSheetItemTableViewCell.m:97]: Is this needed? I see this as the only one with priority?

File: components/ActionSheet/tests/snapshot/MDCActionSheetControllerSnapshotTests.m [hunk 2]
```diff
@@ -370,6 +370,138 @@ - (void)testFiveActionsTooNarrowRTL {
   [self generateSnapshotAndVerifyForView:controller.view];
 }
 
+#pragma mark - contentEdgeInsets
```
Comment (codeman7) [inline components/ActionSheet/tests/snapshot/MDCActionSheetControllerSnapshotTests.m:373]: Can we make a new file for these tests and for the follow up tests `titleEdgeInsets` `imageEgdeInsets` so one file for 3 kinds of test `content/title/image edgeInsets`?

**Review by codeman7 — APPROVED** [review]
In future PRs can we try to break them up? I see this as 3/4 changes. 1. Refactor the cell to have a container for the content 2. New API and test for that API 3. Docs 4. Example. We don't have to break them up in that order but this seems like multiple changes in one PR. Just my opinion I can see how this is _one_ change and they should all go in or be reverted together, I'd love to hear your thoughts on this.

File: components/ActionSheet/src/private/MDCActionSheetItemTableViewCell.m (hunk 1 above)
Comment (romoore) [reply components/ActionSheet/src/private/MDCActionSheetItemTableViewCell.m:97]: Lowering the priority is required to not break constraints when setting content edge insets. If this is kept at `.required`, then the distance from the label to bottom is incompatible.

**Comment by romoore** [comment]
@codeman7 I believe @jverkoey pointed out a PR that had all the changes, tests, docs, examples smushed together as being exemplary. I was attempting to follow that example by smushing these all in one PR.

File: components/ActionSheet/tests/snapshot/MDCActionSheetControllerSnapshotTests.m (hunk 2 above)
Comment (romoore) [reply components/ActionSheet/tests/snapshot/MDCActionSheetControllerSnapshotTests.m:373]: Why do you feel the tests should go in another file? I thought it was nice not to duplicate all the helpers/config methods into another test file.

**Comment by romoore** [comment]
@codeman7, I can split out the container view to another PR, but I don't believe the example/docs should be split. As you mentioned, those go together in case of a roll back.

**Comment by romoore** [comment]
@codeman7 I'm going to merge this as-is based on your offline comment last night, but I can move the tests to a separate file in a follow-up PR. Let's discuss the test locations offline. Let's also clarify expectations with @jverkoey so we are all aligned on what should be in a single PR versus when we should split.

**Comment by jverkoey** [comment]
> I believe @jverkoey pointed out a PR that had all the changes, tests, docs, examples smushed together as being exemplary. I was attempting to follow that example by smushing these all in one PR.

I do consider a PR that includes the code, tests, and relevant documentation to be exemplary. I do, however, agree with Cody that the internal refactoring would have been a good first PR to land and that the public API + docs + example would have been better as a follow-up PR.

> 1. Refactor the cell to have a container for the content 2. New API and test for that API 3. Docs 4. Example

I'd have recommended:

1. Refactor the cell to have a container for the content
2. New API and test for that API + docs
4. Example

**Pull request merged**